corehq/apps/builds/utils.py

```diff
@@ -40,3 +40,47 @@ def extract_build_info_from_filename(content_disposition):
     else:
         raise ValueError('Could not find filename like {!r} in {!r}'.format(
             pattern, content_disposition))
+
+
+def get_latest_version_at_time(target_time):
+    """
+    Get the latest CommCare version that was available at a given time.
+    Excludes superuser-only versions.
+    Menu items are already in chronological order (newest last).
+    If no target time is provided, return the latest version at now.
+    """
+    config = CommCareBuildConfig.fetch()
+
+    if not target_time:
+        return config.get_default().version
+
+    # Iterate through menu items in reverse (newest to oldest)
+    for item in reversed(config.menu):
+        if item.superuser_only:
+            continue
+        try:
+            build = CommCareBuild.get_build(item.build.version, latest=True)
+            if build and build.time and build.time <= target_time:
+                return item.build.version
+        except KeyError:
+            continue
+
+    return None
+
+
+def _parse_version(version_str):
+    """Convert version string to comparable tuple"""
+    if version_str:
+        try:
+            return tuple(int(n) for n in version_str.split('.'))
+        except (ValueError, AttributeError):
+            return None
+    return None
```

Comment:
Nit: move beneath the function that uses it

Reply:
a44b2002dac682103c2027238e0a5111ca72427d